src/app/fyle/view-team-report/view-team-report.page.ts

```
@@ -383,14 +385,18 @@ export class ViewTeamReportPage {
 
       if (data && data.action === 'approve') {
-        this.approverReportsService.approve(report.id).subscribe(() => {
-          this.router.navigate(['/', 'enterprise', 'team_reports']);
-          this.launchDarklyService.getVariation('nps_survey', false).subscribe((showNpsSurvey) => {
-            if (showNpsSurvey) {
-              this.refinerService.startSurvey({ actionName: 'Approve Report' });
-            }
+        this.approveReportLoader = true;
+        this.approverReportsService
+          .approve(report.id)
+          .pipe(finalize(() => (this.approveReportLoader = false)))
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)